### PR TITLE
Add start any number on first unit (v2)

### PR DESCRIPTION
**Pull request by dc-admin**
Hi!
I fix bugs from previous version, and wrote new patch for adding "start any number on first unit" function.

Default settings:
scr1

Cabinet settings:
scr5

Cabinet with bottom to top numeration:
scr2

Cabinet with top to bottom numeration:
scr3

Editing device:
scr4

Regards.

### CODE REVIEW

File: misc.inc.php [hunk 1]
```diff
@@ -923,16 +923,18 @@ function BuildCabinet($cabid,$face="front"){
 	$dev->ParentDevice=0;
 	$bounds=array(
 		'max'=>array('position'=>0,'height'=>0),
-		'min'=>array('position'=>0,'height'=>0),
+		'min'=>array('position'=>$cab->StartUNum,'height'=>0),
 	);
 
 	// Read in all the devices and make sure they fit the cabinet.  If not expand it
 	foreach($dev->Search() as $device){
-		if($device->Position==0){
```
Comment (wilpig) [inline misc.inc.php]: What about people that defined their device position to be -10 for some crazy reason, this is going to skip it and not adjust the rack height to start at -10

**Comment by samilliken** [comment]
You need to present a compelling reason why this is necessary.  There are things that are possible in a data center that we intentionally don't allow in openDCIM because it's simply a very bad practice - the first that comes to mind is allowing devices in one rack to be connected to a power strip in another.  Right now I don't know of a good reason to include this in the main project.

File: scripts/common.js [hunk 2]
```diff
@@ -1350,8 +1350,30 @@ function initdrag(){
...
+        async: false
+    });
+    return result;
+};
+
+
 function InsertDevice(obj){
-	if(obj.Position!=0){
+	var devMinPosition=getMinUnit(obj.Cabinet);
+	if(obj.Position>=devMinPosition){
```
Comment (wilpig) [inline scripts/common.js]: This is also adding an additional call to the system for every device insertion into a rack.  That seems a little excessive.

File: misc.inc.php (hunk 1 above)
Comment (dc-admin) [inline misc.inc.php]: No, this need for visible errors and for devices draggable out of borders rows.

Example:
scr6

**Comment by dc-admin** [comment]
For example, single numbering plan from the floor for racks and cabinets for hanging and marking patch-panels as number NN-XX-YY-UU, NN - the serial number of the structure, XX - the serial number of the server-room, YY - the serial number vertical row of the server-room, UU - the serial number unit from the floor.

File: scripts/common.js (hunk 2 above)
Comment (dc-admin) [inline scripts/common.js]: This function needs for correct dragging devices, if this out of borders cabinet.

**Comment by samilliken** [comment]
Removing stale request - unable to merge with current code base

**Pull request closed without merging**